doc/source/notebooks/advanced/varying_noise.pct.py

```diff
@@ -114,10 +115,12 @@ def variational_expectations(self, Fmu, Fvar, Y):
 def objective_closure():
     return - model.log_marginal_likelihood()
 
-opt = gpflow.optimizers.Scipy()
-opt.minimize(objective_closure,
-             model.trainable_variables,
-             options=dict(maxiter=ci_niter(1000)))
+natgrad_opt = NaturalGradient(gamma=1.0)
+scipy_opt = gpflow.optimizers.Scipy()
+
+for _ in range(ci_niter(1000)):
+    scipy_opt.minimize(objective_closure, model.kernel.trainable_variables, options={'maxiter': 1})
```

Comment:
I think it would be more useful in this example to pass `model.trainable_variables` to the optimiser and explicitly do `set_trainable(model.q_mu, False)` and `set_trainable(model.q_sqrt, False)` for the natgrad optimiser. This to make sure that the natgrad and scipy optimiser are not acting on the same variables and to allow scipy to optimise over all the non-variational parameters.

Reply:
do you mean do all the scipy optimization first, then do the natgrad optimization, or to
```
model.q_mu.set_trainable(False)
model.q_srt.set_trainable(False)
scipy_opt.optimize(..., model.trainable_variables)
model.q_mu.set_trainable(True)
model.q_srt.set_trainable(True)
natgrad_opt.optimize(..., [(model.q_mu, model.q_sqrt)])

Reply:
Typically you want to interleave training of "normal" parameters and hyper-parameters as follows:
```python
model.q_mu.set_trainable(False)
model.q_srt.set_trainable(False)
for i in range(iters):
    scipy_opt.optimize(..., model.trainable_variables)
    natgrad_opt.optimize(..., [(model.q_mu, model.q_sqrt)])
```
The reason we set `q_mu` and `q_sqrt` to non trainable is for the scipy optmiser to ignore them. Even in this state, will the natgrad optimiser still optimise them.

Reply:
ah yes, I forgot that `.trainable_variables` denoted whether they're marked as trainable, not whether they are actually mutable

Reply:
i notice you've omitted maxiter of 1. was that for brevity or cos it's not needed?

Reply:
it's worth mentioning that `set_trainable` reads to me as "this value can't be mutated" rather than "this value won't be gathered by `.trainable_variables`

Reply:
Good catch - you do want `maxiter=1` as I believe the default behaviour of scipy would be to run until convergence.